include/zephyr/net/socket_service.h

```diff
@@ -0,0 +1,246 @@
+/**
+ * @file
+ * @brief BSD Socket service API
+ *
+ * API can be used to install a k_work that is called
+ * if there is data received to a socket.
+ */
+
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_NET_SOCKET_SERVICE_H_
+#define ZEPHYR_INCLUDE_NET_SOCKET_SERVICE_H_
+
+/**
+ * @brief BSD socket service API
+ * @defgroup bsd_socket_service BSD socket service API
+ * @ingroup networking
+ * @{
+ */
+
+#include <sys/types.h>
+#include <zephyr/types.h>
+#include <zephyr/net/socket.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * This struct contains information which socket triggered
+ * calls to the callback function.
+ */
+struct net_socket_service_event {
+	/** k_work that is done when there is desired activity in file descriptor. */
+	struct k_work work;
+	/** Callback to be called for desired socket activity */
+	k_work_handler_t callback;
+	/** Socket information that triggered this event. */
+	struct zsock_pollfd event;
+	/** User data */
+	void *user_data;
+	/** Service back pointer */
+	struct net_socket_service_desc *svc;
+};
+
+/**
+ * Socket service is a struct that user needs to fill properly.
```

Comment:
I changed the first sentence to `Main structure holding socket service configuration information.` as the state information is not stored in this struct (because it is stored in rom).